omod/src/main/java/org/openmrs/module/idgen/rest/resource/IdentifierSourceResource.java

```diff
@@ -0,0 +1,306 @@
+                Object sourceUuid = postBody.get("sourceUuid");
+                Object sequential = postBody.get("sequential");
+                Object refillWithScheduledTask = postBody.get("refillWithScheduledTask");
+
+                if (sourceUuid == null || StringUtils.isBlank(sourceUuid.toString())) {
+                    errors.add("identifier souce");
+                }
+                else{
+                    IdentifierSource poolIdentifierSource = Context.getService(IdentifierSourceService.class).getIdentifierSourceByUuid(sourceUuid.toString());
+                    if(poolIdentifierSource != null){
+                        identifierSource.setSource(poolIdentifierSource);
+                    }else{
+                        errors.add("identifier souce");
+                    }
+                }
+                if (description != null && StringUtils.isNotBlank(description.toString())) {
+                    identifierSource.setDescription(description.toString());
+                }
+                if (batchSize != null && StringUtils.isNotBlank(batchSize.toString())) {
+                    if(StringUtils.isNumeric(batchSize.toString())) {
+                        identifierSource.setBatchSize(Integer.parseInt(batchSize.toString()));
+                    }
+                }
+                if (minPoolSize != null && StringUtils.isNotBlank(minPoolSize.toString())) {
+                    if(StringUtils.isNumeric(minPoolSize.toString())) {
+                        identifierSource.setMinPoolSize(Integer.parseInt(minPoolSize.toString()));
+                    }
+                }
+                if (sequential != null && StringUtils.isNotBlank(sequential.toString())) {
+                    if(parseBoolean(sequential.toString()) != null) {
+                        identifierSource.setSequential(parseBoolean(sequential.toString()));
+                    }
+                }
+                if (refillWithScheduledTask != null && StringUtils.isNotBlank(refillWithScheduledTask.toString())) {
+                    if(parseBoolean(refillWithScheduledTask.toString()) != null) {
+                        identifierSource.setSequential(parseBoolean(refillWithScheduledTask.toString()));
+                    }
+                }
+                if(errors.size() > 0) {
+                    throw new ValidationException("The values of the following inputs are missing or invalid: " + errors.toString());
+                }
+
+                identifierSource.setIdentifierType(patientIdentifierType);
+                identifierSource.setName(name.toString());
+                savedIdentifierSource =  save(identifierSource);
+            }
+            return ConversionUtil.convertToRepresentation(savedIdentifierSource, Representation.DEFAULT);
+        }
+        catch(NullPointerException nullPointerException){
+            throw new ValidationException(nullPointerException.toString());
```

Comment:
What causes this null pointer exception?

Reply:
In the swagger documentation it would just show `[null]` when a required variable is missing and not proceed to the ValidationError() message. Now I am able to show the null pointer error whenever there is a null pointer error.

Reply:
Why would you miss a required variable if you need proper swagger docs?

Reply:
It was intentional to see the error messages

Reply:
Which line throws the NPE? And which variable is it that is null?

Reply:
I uploaded the new omod without the try{} catch{} and it shows the validation error. Really Strange. I have removed the try{} catch{}